frontend/components/TextInput/TextInput.jsx

```diff
@@ -104,21 +126,35 @@ export default class TextInput extends Component {
     this.state.value = props.value || ''
   }
 
+  componentDidMount() {
+    this.adjustHeightIfNeeded()
+  }
+
   render() {
     const {
       className,
+      heightType,
       id,
       inLabel,
       placeholder,
       readonly,
       required,
+      resizable,
+      rows,
       type,
       value
     } = this.props
 
     let inputStyle = new Style(styles, 'input')
 
     inputStyle.addIf('input-in-label', inLabel)
+    inputStyle.add(resizable ? `resizable` : `not-resizable`)
+    if (heightType === `full`) {
+      inputStyle.add(`full-height`)
+    }
+    else if (heightType === `content`) {
```

Comment:
`else` in the same line as `if`, please 😇

Reply:
Oh you mean `} else  if (` ? Of course, missed that one.